darwin: gpgme build broken

While reviewing #70990 I noticed, the bug on a downstream dependency

**Describe the bug**
tries to do a dlopen, and can't find the dynamic library.

**Expected behavior**
Not crash.

**Additional context**
<details> <summary> Fail log </summary>

```
Traceback (most recent call last):
  File "./final.py", line 24, in <module>
    import support
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/tests/support.py", line 27, in <module>
    import gpg
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/__init__.py", line 123, in <module>
    from . import core
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/core.py", line 10, in <module>
    from . import gpgme
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/gpgme.py", line 28, in <module>
    _gpgme = swig_import_helper()
  File "/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/gpgme.py", line 24, in swig_import_helper
    _mod = imp.load_module('_gpgme', fp, pathname, description)
  File "/nix/store/611gq7s9s11w5ri95fbq2adi9ifrhkmn-python3-3.7.4/lib/python3.7/imp.py", line 242, in load_module
    return load_dynamic(name, filename, file)
  File "/nix/store/611gq7s9s11w5ri95fbq2adi9ifrhkmn-python3-3.7.4/lib/python3.7/imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: dlopen(/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/_gpgme.cpython-37m-darwin.so, 2): Library not loaded: /nix/store/95ksx1ywkpp1id8k3xqp9x219r0fjj0k-gpgme-1.13.1/lib/libgpgme.11.dylib
  Referenced from: /private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/python3.7-gpg/lib.macosx-10.6-x86_64-3.7/gpg/_gpgme.cpython-37m-darwin.so
  Reason: image not found
Running tests using /nix/store/611gq7s9s11w5ri95fbq2adi9ifrhkmn-python3-3.7.4/bin/python3.7 (3.7)...
FAIL: initial.py
FAIL: t-wrapper.py
FAIL: t-callbacks.py
FAIL: t-data.py
FAIL: t-encrypt.py
FAIL: t-encrypt-sym.py
FAIL: t-encrypt-sign.py
FAIL: t-sign.py
FAIL: t-signers.py
FAIL: t-decrypt.py
FAIL: t-verify.py
FAIL: t-decrypt-verify.py
FAIL: t-sig-notation.py
FAIL: t-export.py
FAIL: t-import.py
FAIL: t-trustlist.py
FAIL: t-edit.py
FAIL: t-keylist.py
FAIL: t-keylist-from-data.py
FAIL: t-wait.py
FAIL: t-encrypt-large.py
FAIL: t-file-name.py
FAIL: t-idiomatic.py
FAIL: t-protocol-assuan.py
FAIL: t-quick-key-creation.py
FAIL: t-quick-subkey-creation.py
FAIL: t-quick-key-manipulation.py
FAIL: t-quick-key-signing.py
FAIL: final.py
29 tests run, 0 succeeded, 29 failed, 0 skipped.
make[3]: *** [Makefile:613: xcheck] Error 29
make[3]: Leaving directory '/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python/tests'
make[2]: *** [Makefile:472: check-recursive] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang/python'
make[1]: *** [Makefile:457: check-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-gpgme-1.13.1.drv-0/gpgme-1.13.1/lang'
make: *** [Makefile:534: check-recursive] Error 1
builder for '/nix/store/3hvi5bzsjb0x1wrdhwqazs8akpqk5qa5-gpgme-1.13.1.drv' failed with exit code 2
cannot build derivation '/nix/store/napnjdcs66z1sagmndjhipv3pjm69yp3-python3.7-pgpy-0.5.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ncwjx5g23yhak4vvvq4qzmzgnkscclq2-python3.7-srht-0.54.2.drv': 1 dependencies couldn't be built
```
</details>

full log: https://logs.nix.ci/?key=nixos/nixpkgs.70990&attempt_id=d202832a-d134-4823-b199-2a1e24e7e4c4

--- a/pkgs/development/libraries/gpgme/default.nix
+++ b/pkgs/development/libraries/gpgme/default.nix
@@ -1,4 +1,5 @@
-{ stdenv, fetchurl, fetchpatch, libgpgerror, gnupg, pkgconfig, glib, pth, libassuan
+{ stdenv, fetchurl, fetchpatch
+, autoreconfHook, libgpgerror, gnupg, pkgconfig, glib, pth, libassuan
 , file, which, ncurses
 , texinfo
 , buildPackages
@@ -33,7 +34,14 @@ stdenv.mkDerivation rec {
       url = "http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gpgme.git;a=patch;h=c4cf527ea227edb468a84bf9b8ce996807bd6992";
       sha256 = "pKL1tvUw7PB2w4FHSt2up4SvpFiprBH6TLdgKxYFC3g=";
     })
-  ];
+    # https://lists.gnupg.org/pipermail/gnupg-devel/2020-April/034591.html
+    (fetchpatch {
+      name = "0001-Fix-python-tests-on-non-Linux.patch";
+      url = "https://lists.gnupg.org/pipermail/gnupg-devel/attachments/20200415/f7be62d1/attachment.obj";
+      sha256 = "00d4sxq63601lzdp2ha1i8fvybh7dzih4531jh8bx07fab3sw65g";
+    })
+    # Disable python tests on Darwin as they use gpg (see configureFlags below)
+  ] ++ lib.optional stdenv.isDarwin ./disable-python-tests.patch;
 
   outputs = [ "out" "dev" "info" ];
   outputBin = "dev"; # gpgme-config; not so sure about gpgme-tool
@@ -42,23 +50,10 @@ stdenv.mkDerivation rec {
     [ libgpgerror glib libassuan pth ]
     ++ lib.optional (qtbase != null) qtbase;
 
-  nativeBuildInputs = [ file pkgconfig gnupg texinfo ]
+  nativeBuildInputs = [ pkgconfig gnupg texinfo autoreconfHook ]
   ++ lib.optionals pythonSupport [ python swig2 which ncurses ];
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
-
-  postPatch =''
-    substituteInPlace ./configure --replace /usr/bin/file ${file}/bin/file
-  ''
-  # Apply only on Darwin to save rebuilds on Linux
-     + lib.optionalString stdenv.isDarwin ''
-    sed -i "s/LD_LIBRARY_PATH/@shlibpath_var@/" lang/python/tests/Makefile.in
-    sed -i "s/ac_subst_vars='/ac_subst_vars='shlibpath_var\n/" configure
-  ''
-  # Disable python tests on Darwin as they use gpg (see configureFlags below)
-     + lib.optionalString stdenv.isDarwin ''
-    sed -i "s/SUBDIRS = \. tests/SUBDIRS = ./" lang/python/Makefile.in
-  '';
 
   configureFlags = [
     "--enable-fixed-path=${gnupg}/bin"

--- a/pkgs/development/libraries/gpgme/default.nix
+++ b/pkgs/development/libraries/gpgme/default.nix
@@ -49,6 +49,15 @@ stdenv.mkDerivation rec {
 
   postPatch =''
     substituteInPlace ./configure --replace /usr/bin/file ${file}/bin/file
+  ''
+  # Apply only on Darwin to save rebuilds on Linux
+     + lib.optionalString stdenv.isDarwin ''
+    sed -i "s/LD_LIBRARY_PATH/@shlibpath_var@/" lang/python/tests/Makefile.in
+    sed -i "s/ac_subst_vars='/ac_subst_vars='shlibpath_var\n/" configure
+  ''
+  # Disable python tests on Darwin as they use gpg (see configureFlags below)
+     + lib.optionalString stdenv.isDarwin ''
+    sed -i "s/SUBDIRS = \. tests/SUBDIRS = ./" lang/python/Makefile.in
   '';
 
   configureFlags = [

--- a/pkgs/development/libraries/gpgme/disable-python-tests.patch
+++ b/pkgs/development/libraries/gpgme/disable-python-tests.patch
@@ -1,0 +1,12 @@
+diff -Naur --strip-trailing-cr gpgme-1.13.1.org/lang/python/Makefile.am gpgme-1.13.1/lang/python/Makefile.am
+--- gpgme-1.13.1.org/lang/python/Makefile.am	2019-06-04 07:27:49.000000000 +0100
++++ gpgme-1.13.1/lang/python/Makefile.am	2020-04-15 14:27:34.810172944 +0100
+@@ -23,7 +23,7 @@
+ 	gpgme.i \
+ 	helpers.c helpers.h private.h
+ 
+-SUBDIRS = . tests examples doc src
++SUBDIRS = . examples doc src
+ 
+ .PHONY: prepare
+ prepare: copystamp

